<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PartGui::PatternParametersWidget</class>
 <widget class="QWidget" name="PartGui::PatternParametersWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>270</width>
    <height>200</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="generalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Direction</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QToolButton" name="PushButtonReverse">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Reverse the direction of the pattern.</string>
          </property>
          <property name="icon">
           <iconset resource="Resources/resource.qrc">
           <normaloff>:/icons/button_sort.svg</normaloff>:/icons/button_sort.svg</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboDirection"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_Mode">
        <item>
         <widget class="QLabel" name="labelMode">
          <property name="text">
           <string>Mode</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboMode">
          <item>
           <property name="text">
            <string>Extent</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Spacing</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QWidget" name="extentWrapper" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_Length">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="labelLength">
           <property name="text">
            <string>Length</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="Gui::QuantitySpinBox" name="spinExtent" native="true">
           <property name="keyboardTracking" stdset="0">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="spacingWrapper" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout_Offset">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="labelOffset">
           <property name="text">
            <string>Spacing</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="Gui::QuantitySpinBox" name="spinSpacing" native="true">
           <property name="keyboardTracking" stdset="0">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="addSpacingButton">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Add spacing to create spacing patterns.</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset>
             <normaloff>:/icons/list-add.svg</normaloff>:/icons/list-add.svg</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="spacingPatternWidget" native="true">
        <layout class="QVBoxLayout" name="spacingPatternLayout">
          <property name="spacing">
           <number>3</number>
          </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_Occurrences">
        <item>
         <widget class="QLabel" name="labelOccurrences">
          <property name="text">
           <string>Occurrences</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Gui::UIntSpinBox" name="spinOccurrences"/>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Gui::QuantitySpinBox</class>
   <extends>QWidget</extends>
   <header>Gui/QuantitySpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>Gui::UIntSpinBox</class>
   <extends>QSpinBox</extends>
   <header>Gui/SpinBox.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>